bug: Incorrect zeros in GPU Inventory History

### Describe the bug

The history sometimes shows that there are no GPUs at all, sometimes shows a more reasonable inventory. The real inventory is not flapping like that.

<img width="1117" height="644" alt="Image" src="https://github.com/user-attachments/assets/12259ba2-c1d2-491d-ad14-c02011b4d8c2" />

### Environment

- Browser: Firefox 149.0
- OS: MacOS 15.7.4
- Console: 0.3.20

### Steps To Reproduce

1. Add the GPU Inventory History card
2. Make it full screen
3. Observe flapping in the displayed history

### Expected Behavior

Consistent accuracy

### Browser Console / Backend Logs

```shell
Browser console tail, Parent process only:

14:09:11.919 NewTabGleanUtils: Failed to complete registration of metrics and pings found in runtime metrics JSON: TypeError: NetworkError when attempting to fetch resource. NewTabGleanUtils.sys.mjs:116:23
09:23:46.650 No response for feed DiscoveryStreamFeed.sys.mjs:1901:17
21:18:02.065 services.settings: NetworkError: Network error: TypeError: NetworkError: Network request failed
    APIError resource://services-settings/RemoteSettingsClient.sys.mjs:92
    NetworkError resource://services-settings/RemoteSettingsClient.sys.mjs:96
    _adjustedError resource://services-settings/RemoteSettingsClient.sys.mjs:991
    maybeSync resource://services-settings/RemoteSettingsClient.sys.mjs:887
Caused by: TypeError: NetworkError: Network request failed
    onerror resource://services-settings/Utils.sys.mjs:254
remote-settings.sys.mjs:547
21:18:02.272 BroadcastService: receivedBroadcastMessage: handler for remote-settings/monitor_changes threw error: NetworkError: Network error: TypeError: NetworkError: Network request failed
    APIError resource://services-settings/RemoteSettingsClient.sys.mjs:92
    NetworkError resource://services-settings/RemoteSettingsClient.sys.mjs:96
    _adjustedError resource://services-settings/RemoteSettingsClient.sys.mjs:991
    maybeSync resource://services-settings/RemoteSettingsClient.sys.mjs:887
Caused by: TypeError: NetworkError: Network request failed
    onerror resource://services-settings/Utils.sys.mjs:254
PushBroadcastService.sys.mjs:255:22
10:31:27.513 NS_ERROR_NOT_AVAILABLE: 
10:31:50.553 NS_ERROR_NOT_AVAILABLE: 
15:18:54.546 NS_ERROR_NOT_AVAILABLE: 
15:53:20.654 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
15:53:21.012 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
15:53:22.120 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
15:53:22.287 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
15:53:23.313 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
15:53:23.381 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
15:53:24.282 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
15:53:24.344 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:31:03.998 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:31:04.112 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:17.977 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:18.082 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:18.186 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:18.719 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:19.015 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:20.736 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:20.858 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:22.770 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:22.871 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:23.042 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:23.142 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:24.528 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:25.009 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:25.118 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:25.181 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:25.292 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:26.200 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:26.957 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:28.577 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:29.243 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:29.319 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
01:53:29.676 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
02:06:28.273 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
03:57:40.890 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
03:57:41.831 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
03:57:41.917 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
03:57:42.142 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
03:57:42.202 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
10:10:43.282 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
10:10:43.362 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
10:10:43.431 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
10:10:46.230 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
10:11:31.614 Cookie warnings 2
12:59:26.311 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
12:59:26.445 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
12:59:26.684 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
12:59:26.806 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
12:59:26.924 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
12:59:28.530 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
12:59:28.895 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
12:59:28.983 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
13:01:41.853 Cookie warnings 15
13:01:42.128 Partitioned cookie or storage access was provided to “<URL>” because it is loaded in the third-party context and dynamic state partitioning is enabled. 8
22:26:30.259 Sync encountered an error - see about:sync-log for the log file. policies.sys.mjs:977:15
23:11:14.806 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
23:11:15.088 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
23:11:15.154 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
23:11:15.665 URLBar - MerinoClient [SuggestBackendMerino]: Fetch error NS_ERROR_NOT_AVAILABLE: Component returned failure code: 0x80040111 (NS_ERROR_NOT_AVAILABLE) [nsIHttpChannel.responseStatus]
    onStopRequest resource://gre/modules/ObliviousHTTP.sys.mjs:182
MerinoClient.sys.mjs:338:31
23:46:55.324 Cookie warnings 32
00:11:49.892 Content-Security-Policy warnings 4
00:58:08.489 Content process 96494 isn't responsive while sending "DevToolsProcessParent:watchTargets" request. DevTools will ignore this process for now. DevToolsProcessParent.sys.mjs:320:17
00:58:08.489 Content process 299 isn't responsive while sending "DevToolsProcessParent:watchTargets" request. DevTools will ignore this process for now. DevToolsProcessParent.sys.mjs:320:17
    timeout resource://devtools/server/connectors/js-process-actor/DevToolsProcessParent.sys.mjs:320
    notify resource://gre/modules/Timer.sys.mjs:66
00:58:08.842 Content process 96494 is responsive again. DevTools resumes operations against it. DevToolsProcessParent.sys.mjs:330:21
00:58:08.995 Content process 299 is responsive again. DevTools resumes operations against it. DevToolsProcessParent.sys.mjs:330:21
01:02:58.110 Cookie warnings 2
01:06:13.401 Content process 86577 isn't responsive while sending "DevToolsProcessParent:addOrSetSessionDataEntry" request. DevTools will ignore this process for now. DevToolsProcessParent.sys.mjs:320:17
01:06:13.401 Content process 42444 isn't responsive while sending "DevToolsProcessParent:addOrSetSessionDataEntry" request. DevTools will ignore this process for now. DevToolsProcessParent.sys.mjs:320:17
01:08:37.465 SyntaxError: JSON.parse: expected double-quoted property name at line 1 column 153 of the JSON data
01:08:37.465 Reporting Header: invalid JSON value received. collect
```

### Want to contribute?

- [ ] I would like to work on this issue.

### Additional Context

_No response_

--- a/web/src/components/cards/GPUInventoryHistory.tsx
+++ b/web/src/components/cards/GPUInventoryHistory.tsx
@@ -4,6 +4,7 @@ import {
   BarChart3, Table2, ChevronDown, ArrowUpDown } from 'lucide-react'
 import ReactECharts from 'echarts-for-react'
 import { useMetricsHistory } from '../../hooks/useMetricsHistory'
+import type { MetricsSnapshot } from '../../types/predictions'
 import { useCachedGPUNodes } from '../../hooks/useCachedData'
 import { useDemoMode } from '../../hooks/useDemoMode'
 import { useGlobalFilters } from '../../hooks/useGlobalFilters'
@@ -461,7 +462,7 @@ export function GPUInventoryHistory() {
       return generateDemoData()
     }
 
-    return (history || []).map(snapshot => {
+    const points = (history || []).map(snapshot => {
       const filtered = filterGPUNodes(snapshot.gpuNodes || [])
       const allocated = filtered.reduce((sum, g) => sum + (g.gpuAllocated || 0), 0)
       const total = filtered.reduce((sum, g) => sum + (g.gpuTotal || 0), 0)
@@ -488,6 +489,19 @@ export function GPUInventoryHistory() {
 
       return point
     })
+
+    // Defensive filter: if any point in the series has a non-zero total, drop
+    // points whose total is zero. A zero-total point next to non-zero points
+    // is almost always a transient GPU-fetch glitch that slipped through
+    // carry-forward protection in useMetricsHistory. Keeps the chart,
+    // stats, and trend math from showing misleading flapping zero bars.
+    // If EVERY point has total === 0 (legitimate no-GPU state) we leave the
+    // series alone so the empty-state paths still fire correctly.
+    const anyNonZero = points.some(p => p.total > 0)
+    if (anyNonZero) {
+      return points.filter(p => p.total > 0)
+    }
+    return points
   }, [history, showDemo, filterGPUNodes, chartMode])
 
   /** All GPU type keys present in chart data */
@@ -559,15 +573,24 @@ export function GPUInventoryHistory() {
 
   // ── Churn metrics ──────────────────────────────────────────────────
   const churnMetrics = useMemo<ChurnMetrics | null>(() => {
-    if (showDemo || (history || []).length < MIN_CHURN_SNAPSHOTS) return null
+    // Drop snapshots with zero total GPUs before diffing so transient empty
+    // captures don't show up as massive departure/arrival churn.
+    const churnHistory = (history || []).filter(s => {
+      const nodes = s.gpuNodes || []
+      if (nodes.length === 0) return false
+      const total = nodes.reduce((sum, g) => sum + (g.gpuTotal || 0), 0)
+      return total > 0
+    })
+
+    if (showDemo || churnHistory.length < MIN_CHURN_SNAPSHOTS) return null
 
     let totalArrivals = 0
     let totalDepartures = 0
     let diffCount = 0
 
-    for (let i = 1; i < (history || []).length; i++) {
-      const prev = filterGPUNodes((history || [])[i - 1].gpuNodes || [])
-      const curr = filterGPUNodes((history || [])[i].gpuNodes || [])
+    for (let i = 1; i < churnHistory.length; i++) {
+      const prev = filterGPUNodes(churnHistory[i - 1].gpuNodes || [])
+      const curr = filterGPUNodes(churnHistory[i].gpuNodes || [])
 
       // Per-node diffing captures churn even when allocations and frees cancel at aggregate level
       const prevMap: Record<string, number> = {}
@@ -631,7 +654,24 @@ export function GPUInventoryHistory() {
   const tableRows = (() => {
     if (showDemo) return generateDemoTableRows()
 
-    const latestSnapshot = (history || []).length > 0 ? (history || [])[(history || []).length - 1] : null
+    // Walk history from the end and pick the most recent snapshot whose
+    // total GPU count is non-zero. Falls back to the literal latest (even if
+    // zero) so genuine no-GPU clusters still render an empty table rather
+    // than a stale one.
+    let latestSnapshot: MetricsSnapshot | null = null
+    const hist = history || []
+    for (let i = hist.length - 1; i >= 0; i--) {
+      const s = hist[i]
+      const nodes = s.gpuNodes || []
+      const total = nodes.reduce((sum, g) => sum + (g.gpuTotal || 0), 0)
+      if (total > 0) {
+        latestSnapshot = s
+        break
+      }
+    }
+    if (!latestSnapshot && hist.length > 0) {
+      latestSnapshot = hist[hist.length - 1]
+    }
     if (!latestSnapshot) return []
 
     const filtered = filterGPUNodes(latestSnapshot.gpuNodes || [])

--- a/web/src/components/cards/__tests__/GPUInventoryHistory.test.tsx
+++ b/web/src/components/cards/__tests__/GPUInventoryHistory.test.tsx
@@ -49,7 +49,8 @@ vi.mock('../../../hooks/useGlobalFilters', () => ({
   useGlobalFilters: () => ({ selectedClusters: [], isAllClustersSelected: true, selectedSeverities: [], isAllSeveritiesSelected: true, customFilter: '' }),
 }))
 
-vi.mock('../../../hooks/useMetricsHistory', () => ({ useMetricsHistory: () => ({ history: [], isLoading: false }) }))
+const mockUseMetricsHistory = vi.fn(() => ({ history: [], isLoading: false }))
+vi.mock('../../../hooks/useMetricsHistory', () => ({ useMetricsHistory: () => mockUseMetricsHistory() }))
 
 import { GPUInventoryHistory } from '../GPUInventoryHistory'
 
@@ -59,6 +60,7 @@ describe('GPUInventoryHistory', () => {
     mockUseDemoMode.mockReturnValue({ isDemoMode: true, toggleDemoMode: vi.fn(), setDemoMode: vi.fn() })
     mockUseCardLoadingState.mockReturnValue({ showSkeleton: false, showEmptyState: false, hasData: true, isRefreshing: false })
     mockGPUNodes.mockReturnValue({ nodes: [], isLoading: false, isRefreshing: false, isDemoFallback: false, isFailed: false, consecutiveFailures: 0, error: null, lastRefresh: Date.now() })
+    mockUseMetricsHistory.mockReturnValue({ history: [], isLoading: false })
   })
 
   it('renders without crashing', () => {
@@ -100,6 +102,29 @@ describe('GPUInventoryHistory', () => {
     mockGPUNodes.mockReturnValue({ nodes: [], isLoading: false, isRefreshing: false, isDemoFallback: true, isFailed: false, consecutiveFailures: 0, error: null, lastRefresh: Date.now() })
     render(<GPUInventoryHistory />)
     expect(mockUseCardLoadingState).toHaveBeenCalled()
+  })
+
+  it('renders without crashing when history contains a zero-total snapshot mixed with real data', () => {
+    // Regression guard for the flapping-zeros bug: a transient empty capture
+    // slipping through carry-forward should be filtered visually so the
+    // component still renders cleanly.
+    mockUseDemoMode.mockReturnValue({ isDemoMode: false, toggleDemoMode: vi.fn(), setDemoMode: vi.fn() })
+    mockGPUNodes.mockReturnValue({
+      nodes: [{ name: 'node-a', cluster: 'cl-1', gpuType: 'H100', gpuAllocated: 2, gpuCount: 8 }],
+      isLoading: false, isRefreshing: false, isDemoFallback: false,
+      isFailed: false, consecutiveFailures: 0, error: null, lastRefresh: Date.now(),
+    })
+    mockUseMetricsHistory.mockReturnValue({
+      history: [
+        { timestamp: new Date(Date.now() - 30 * 60 * 1000).toISOString(), clusters: [], podIssues: [], gpuNodes: [{ name: 'node-a', cluster: 'cl-1', gpuType: 'H100', gpuAllocated: 2, gpuCount: 8, gpuTotal: 8, gpuAllocated_: 2 }] },
+        // Transient zero snapshot — must be filtered out of chart/stats
+        { timestamp: new Date(Date.now() - 20 * 60 * 1000).toISOString(), clusters: [], podIssues: [], gpuNodes: [] },
+        { timestamp: new Date(Date.now() - 10 * 60 * 1000).toISOString(), clusters: [], podIssues: [], gpuNodes: [{ name: 'node-a', cluster: 'cl-1', gpuType: 'H100', gpuAllocated: 3, gpuCount: 8, gpuTotal: 8, gpuAllocated_: 3 }] },
+      ],
+      isLoading: false,
+    })
+    const { container } = render(<GPUInventoryHistory />)
+    expect(container).toBeTruthy()
   })
 
 })

--- a/web/src/hooks/__tests__/useMetricsHistory.test.ts
+++ b/web/src/hooks/__tests__/useMetricsHistory.test.ts
@@ -789,6 +789,75 @@ describe('useMetricsHistory', () => {
     })
   })
 
+  // ── GPU carry-forward protection (issue referenced in commit/PR body) ──
+  describe('GPU carry-forward protection for transient empty fetches', () => {
+    it('carries forward last known gpuNodes when a single capture has empty GPUs', async () => {
+      // Seed with a snapshot that has real GPU nodes
+      mockClusters.push({ name: 'c1', cpuCores: 4, cpuUsageCores: 2, memoryGB: 8, memoryUsageGB: 4, nodeCount: 1, healthy: true })
+      mockGPUNodes.push({ name: 'gpu-node-1', cluster: 'c1', gpuType: 'NVIDIA H100', gpuAllocated: 3, gpuCount: 8 })
+
+      const { useMetricsHistory } = await importFresh()
+      const { result } = renderHook(() => useMetricsHistory())
+
+      // First capture: real GPU data
+      act(() => { result.current.captureNow() })
+      const firstLatest = result.current.history[result.current.history.length - 1]
+      expect(firstLatest.gpuNodes).toHaveLength(1)
+      expect(firstLatest.gpuNodes[0].gpuTotal).toBe(8)
+
+      // Simulate transient fetch failure: gpuNodes list goes empty
+      mockGPUNodes.length = 0
+
+      // Second capture: should carry forward the previous gpuNodes
+      act(() => { result.current.captureNow() })
+      const secondLatest = result.current.history[result.current.history.length - 1]
+      expect(secondLatest.gpuNodes).toHaveLength(1)
+      expect(secondLatest.gpuNodes[0].name).toBe('gpu-node-1')
+      expect(secondLatest.gpuNodes[0].gpuTotal).toBe(8)
+    })
+
+    it('eventually accepts empty GPU state after the carry-forward window expires', async () => {
+      // MAX_GPU_CARRY_FORWARD = 2 — after two empty captures, the third must reflect the empty state.
+      mockClusters.push({ name: 'c1', cpuCores: 4, cpuUsageCores: 2, memoryGB: 8, memoryUsageGB: 4, nodeCount: 1, healthy: true })
+      mockGPUNodes.push({ name: 'gpu-node-1', cluster: 'c1', gpuType: 'NVIDIA H100', gpuAllocated: 3, gpuCount: 8 })
+
+      const { useMetricsHistory } = await importFresh()
+      const { result } = renderHook(() => useMetricsHistory())
+
+      // Capture with real GPU data
+      act(() => { result.current.captureNow() })
+
+      // Now transition to empty GPU state
+      mockGPUNodes.length = 0
+
+      // Two empty captures — both should be carried-forward
+      act(() => { result.current.captureNow() })
+      act(() => { result.current.captureNow() })
+      const afterCarryForward = result.current.history[result.current.history.length - 1]
+      expect(afterCarryForward.gpuNodes).toHaveLength(1)
+
+      // Third consecutive empty capture — must accept the empty state
+      act(() => { result.current.captureNow() })
+      const afterWindow = result.current.history[result.current.history.length - 1]
+      expect(afterWindow.gpuNodes).toHaveLength(0)
+    })
+
+    it('does not carry-forward when previous snapshot also had empty gpuNodes', async () => {
+      // Non-GPU cluster: every capture has empty gpuNodes — must not carry forward.
+      mockClusters.push({ name: 'c1', cpuCores: 4, cpuUsageCores: 2, memoryGB: 8, memoryUsageGB: 4, nodeCount: 1, healthy: true })
+      // mockGPUNodes intentionally left empty
+
+      const { useMetricsHistory } = await importFresh()
+      const { result } = renderHook(() => useMetricsHistory())
+
+      act(() => { result.current.captureNow() })
+      act(() => { result.current.captureNow() })
+
+      const latest = result.current.history[result.current.history.length - 1]
+      expect(latest.gpuNodes).toHaveLength(0)
+    })
+  })
+
   describe('trend edge cases', () => {
     it('getClusterTrend returns "stable" for a non-existent cluster', async () => {
       const snaps = [

--- a/web/src/hooks/useMetricsHistory.ts
+++ b/web/src/hooks/useMetricsHistory.ts
@@ -10,10 +10,24 @@ const MAX_SNAPSHOTS = 1008 // 7 days at 10-min intervals (6 per hour * 24 hours 
 const CACHE_TTL_MS = 7 * 24 * 60 * 60 * 1000
 /** Maximum number of increasing-restart pods to include in AI context */
 const MAX_INCREASING_RESTART_PODS = 10
+/**
+ * Maximum consecutive snapshots whose empty GPU data will be carried-forward
+ * from the last known non-empty gpuNodes list. Prevents a transient GPU fetch
+ * glitch (SSE race, partial cluster reachability) from being persisted as a
+ * zero-total snapshot in GPU Inventory History while still allowing truly
+ * removed GPUs to reflect after ~2 intervals.
+ */
+const MAX_GPU_CARRY_FORWARD = 2
 
 // Singleton state - shared across all hook instances
 let snapshots: MetricsSnapshot[] = []
 const subscribers = new Set<(snapshots: MetricsSnapshot[]) => void>()
+/**
+ * Tracks how many consecutive captures have had an empty gpuNodes list while
+ * the last persisted snapshot still had GPU inventory. Used to carry-forward
+ * the last known gpuNodes for up to MAX_GPU_CARRY_FORWARD captures.
+ */
+let consecutiveEmptyGPUCaptures = 0
 
 // Initialize from localStorage
 if (typeof window !== 'undefined') {
@@ -97,6 +111,51 @@ function addSnapshot(snapshot: MetricsSnapshot) {
 }
 
 /**
+ * Applies carry-forward protection against transient empty gpuNodes results.
+ *
+ * When the current capture has an empty gpuNodes list but the most recent
+ * persisted snapshot still had GPU inventory, replace the empty list with the
+ * previous snapshot's gpuNodes — up to MAX_GPU_CARRY_FORWARD consecutive
+ * captures. This prevents transient fetch glitches from being persisted as
+ * zero-total snapshots that then render as flapping zero bars in GPU
+ * Inventory History.
+ *
+ * Returns the gpuNodes to use for the new snapshot.
+ */
+function applyGPUCarryForward(
+  currentGpuNodes: MetricsSnapshot['gpuNodes'],
+): MetricsSnapshot['gpuNodes'] {
+  // Non-empty current: no carry-forward needed, reset counter.
+  if (currentGpuNodes.length > 0) {
+    consecutiveEmptyGPUCaptures = 0
+    return currentGpuNodes
+  }
+
+  // Empty current. Look at the most recent persisted snapshot.
+  const lastSnapshot = snapshots.length > 0 ? snapshots[snapshots.length - 1] : null
+  const lastHadGPUs = !!lastSnapshot && (lastSnapshot.gpuNodes || []).length > 0
+
+  // No previous GPU data to carry forward — this is a legitimate "no GPUs"
+  // state (either a non-GPU cluster or the zero has already been persisted).
+  if (!lastHadGPUs) {
+    consecutiveEmptyGPUCaptures = 0
+    return currentGpuNodes
+  }
+
+  // Previous snapshot had GPUs, current is empty — likely a transient flap.
+  // Carry forward the previous gpuNodes, but only for a bounded number of
+  // consecutive captures so truly-removed GPUs eventually reflect in history.
+  if (consecutiveEmptyGPUCaptures < MAX_GPU_CARRY_FORWARD) {
+    consecutiveEmptyGPUCaptures += 1
+    return lastSnapshot.gpuNodes
+  }
+
+  // Exceeded the carry-forward window — accept the empty state as truth.
+  consecutiveEmptyGPUCaptures = 0
+  return currentGpuNodes
+}
+
+/**
  * Hook to manage metrics history for trend detection
  * Automatically captures snapshots every 10 minutes (configurable)
  */
@@ -176,6 +235,13 @@ export function useMetricsHistory() {
         return
       }
 
+      const mappedGpuNodes = (currentGpuNodes || []).map(g => ({
+        name: g.name,
+        cluster: g.cluster,
+        gpuType: g.gpuType || '',
+        gpuAllocated: g.gpuAllocated,
+        gpuTotal: g.gpuCount }))
+
       const snapshot: MetricsSnapshot = {
         timestamp: new Date().toISOString(),
         clusters: currentClusters.map(c => ({
@@ -190,12 +256,9 @@ export function useMetricsHistory() {
           cluster: p.cluster || '',
           restarts: p.restarts || 0,
           status: p.status || '' })),
-        gpuNodes: (currentGpuNodes || []).map(g => ({
-          name: g.name,
-          cluster: g.cluster,
-          gpuType: g.gpuType || '',
-          gpuAllocated: g.gpuAllocated,
-          gpuTotal: g.gpuCount })) }
+        // Apply carry-forward to smooth transient empty-GPU fetch results
+        // (see applyGPUCarryForward docstring).
+        gpuNodes: applyGPUCarryForward(mappedGpuNodes) }
 
       addSnapshot(snapshot)
       lastSnapshotRef.current = now
@@ -221,6 +284,13 @@ export function useMetricsHistory() {
   const captureNow = () => {
     if (clusters.length === 0) return
 
+    const mappedGpuNodes = (gpuNodes || []).map(g => ({
+      name: g.name,
+      cluster: g.cluster,
+      gpuType: g.gpuType || '',
+      gpuAllocated: g.gpuAllocated,
+      gpuTotal: g.gpuCount }))
+
     const snapshot: MetricsSnapshot = {
       timestamp: new Date().toISOString(),
       clusters: clusters.map(c => ({
@@ -234,12 +304,7 @@ export function useMetricsHistory() {
         cluster: p.cluster || '',
         restarts: p.restarts || 0,
         status: p.status || '' })),
-      gpuNodes: (gpuNodes || []).map(g => ({
-        name: g.name,
-        cluster: g.cluster,
-        gpuType: g.gpuType || '',
-        gpuAllocated: g.gpuAllocated,
-        gpuTotal: g.gpuCount })) }
+      gpuNodes: applyGPUCarryForward(mappedGpuNodes) }
 
     addSnapshot(snapshot)
     lastSnapshotRef.current = Date.now()
@@ -248,6 +313,7 @@ export function useMetricsHistory() {
   // Clear history
   const clearHistory = () => {
     snapshots = []
+    consecutiveEmptyGPUCaptures = 0
     notifySubscribers()
     persistSnapshots()
   }

